core/templates/dev/head/app.constants.ajs.ts

```diff
@@ -0,0 +1,152 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Shared constants for the Oppia module.
+ */
+
+import { AppConstants } from 'app.constants.ts';
+
+var oppia = require('AppInit.ts').module;
+
+for (var constantName in constants) {
+  oppia.constant(constantName, constants[constantName]);
+}
+
+// Translations of strings that are loaded in the front page. They are listed
+// here to be loaded synchronously with the script to prevent a FOUC or
+// Flash of Untranslated Content.
+// See http://angular-translate.github.io/docs/#/guide/12_asynchronous-loading
+oppia.constant('DEFAULT_TRANSLATIONS', AppConstants.DEFAULT_TRANSLATIONS);
```

Comment:
This seems very boilerplate-y, given that we already enumerate the constants in app.constants.ts. Can we do it more "automatically" instead? Otherwise, we're introducing room for failure here since a dev needs to modify two locations when adding one constant.

Reply:
I saw the need to do this [here](https://www.competa.com/blog/migrate-angularjs-to-angular-using-upgrademodule-part-2/). 

Having this done "automatically" (maybe during build.py) is somewhat complicated. If you like, we can have a lint check to check that a constant is defined in the corresponding ajs file.

Reply:
Thanks for the reference!

I would suggest three things:

- Have a lint check for the transition period to make sure the constants stay in sync.
- Include a description of this information on a "how to use Angular" page in the wiki. This page should contain info for developers who are having issues with Angular in general and need to understand how it works (similar to our wiki page for webpack).
- File an issue and self-assign to get rid of the relevant version once the migration is complete. As you continue doing the migration and adding additional interim stuff, add new checkboxes to that issue with things to remove, so that we have all that info in one place when it comes time to complete the migration.

Does that sound good?

Reply:
Yeah that sounds pretty good. I'll do that!
Thanks!

Reply:
Added the lint check. PTAL!

Reply:
The other bullet points should also be completed before this PR is merged.

Reply:
The wiki page -> https://github.com/oppia/oppia/wiki/How-to-use-Angular

Reply:
The issue -> https://github.com/oppia/oppia/issues/7092

Reply:
Thanks!

In your wiki page, could you please give some code-snippet examples or point to examples in the codebase? It would be clearer that way.

Reply:
I have added them! PTAL.

Reply:
Much better :D Thanks!